### PR TITLE
Fixing rule permission bug

### PR DESCRIPTION
**Pull request by freddyucv**
Fixing: https://github.com/dotCMS/core/issues/17901#issuecomment-591155814

- Using the host object allow to check the permission correctly

https://github.com/dotCMS/core/compare/issue-17901-Unable-to-create-rules?expand=1#diff-8861622cb5192fca78268f8d23179f81R89

- Not catch the DotSecurityException to let the DotSecurityExceptionMapper  catch it, and response with the right format, this avoid a blank page when the user does not have permission

https://github.com/dotCMS/core/compare/issue-17901-Unable-to-create-rules?expand=1#diff-8861622cb5192fca78268f8d23179f81L231

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/sites/ruleengine/rules/RuleResource.java [hunk 1]
```diff
@@ -213,26 +216,19 @@ private User getUser(final HttpServletRequest request, final HttpServletResponse
         return webResource.init(request, response, true).getUser();
     }
 
-    private List<RestRule> getRulesInternal(User user, Ruleable host) {
-        try {
-            List<RestRule> restRules = new ArrayList<>();
+    private List<RestRule> getRulesInternal(User user, Ruleable host) throws DotSecurityException, DotDataException {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/sites/ruleengine/rules/RuleResource.java]: Codacy Issue found: [Parameter 'user' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5093298)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/sites/ruleengine/rules/RuleResource.java [hunk 2]
```diff
@@ -77,11 +79,15 @@ protected RuleResource(ApiProvider apiProvider, WebResource webResource) {
     @Path("/rules")
     @NoCache
     @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
-    public Map<String, RestRule> list(@Context HttpServletRequest request, @Context final HttpServletResponse response, @PathParam("siteId") String siteId) {
+    public Map<String, RestRule> list(
+            @Context HttpServletRequest request,
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/sites/ruleengine/rules/RuleResource.java]: Codacy Issue found: [Parameter 'request' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5093298)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 3]
```diff
@@ -7,10 +7,7 @@
 import com.dotcms.util.IntegrationTestInitService;
 import com.dotmarketing.beans.Host;
 import com.dotmarketing.beans.Permission;
-import com.dotmarketing.business.APILocator;
-import com.dotmarketing.business.CacheLocator;
-import com.dotmarketing.business.PermissionAPI;
-import com.dotmarketing.business.Role;
+import com.dotmarketing.business.*;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.business'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5093298)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/sites/ruleengine/rules/RuleResource.java [hunk 4]
```diff
@@ -77,11 +79,15 @@ protected RuleResource(ApiProvider apiProvider, WebResource webResource) {
...
+            @Context final HttpServletResponse response,
+            @PathParam("siteId") String siteId) throws DotSecurityException, DotDataException {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/sites/ruleengine/rules/RuleResource.java]: Codacy Issue found: [The String literal "siteId" appears 5 times in this file; the first occurrence is on line 85](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5093298)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 5]
```diff
@@ -52,6 +51,74 @@ public static void prepare() throws Exception {
...
+     *
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test (expected = DotSecurityException.class)
+    public void shouldThrowDotSecurityException() throws DotSecurityException, DotDataException {
+        final Role role = new RoleDataGen().nextPersisted();
+        final User user = new UserDataGen().roles(role).nextPersisted();
+        final Host host = new SiteDataGen().nextPersisted();
+        final Rule rule1 = new RuleDataGen().host(host).nextPersisted();
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [Avoid unused local variables such as 'rule1'.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5093298)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/sites/ruleengine/rules/RuleResource.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/sites/ruleengine/rules/RuleResource.java]: Codacy Issue found: [Avoid reassigning parameters such as 'siteId'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5093298)

**Review by jdotcms — APPROVED** [review]

**Review by nollymar — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/sites/ruleengine/rules/RuleResource.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/sites/ruleengine/rules/RuleResource.java]: Codacy Issue found: [Avoid reassigning parameters such as 'siteId'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5093298)

Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/sites/ruleengine/rules/RuleResource.java]: Codacy Issue found: [The String literal "siteId" appears 5 times in this file; the first occurrence is on line 85](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5093298)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.business'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5093298)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/sites/ruleengine/rules/RuleResource.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/sites/ruleengine/rules/RuleResource.java]: Codacy Issue found: [Parameter 'request' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5093298)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/sites/ruleengine/rules/RuleResource.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/sites/ruleengine/rules/RuleResource.java]: Codacy Issue found: [Parameter 'user' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5093298)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [Avoid unused local variables such as 'rule1'.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5093298)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/sites/ruleengine/rules/RuleResource.java [hunk 6]
```diff
@@ -77,11 +79,15 @@ protected RuleResource(ApiProvider apiProvider, WebResource webResource) {
     @Path("/rules")
     @NoCache
     @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
-    public Map<String, RestRule> list(@Context HttpServletRequest request, @Context final HttpServletResponse response, @PathParam("siteId") String siteId) {
+    public Map<String, RestRule> list(
+            @Context final HttpServletRequest request,
+            @Context final HttpServletResponse response,
+            @PathParam("siteId") String siteId) throws DotSecurityException, DotDataException {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/sites/ruleengine/rules/RuleResource.java]: Codacy Issue found: [Avoid reassigning parameters such as 'siteId'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5093298)

Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/sites/ruleengine/rules/RuleResource.java]: Codacy Issue found: [The String literal "siteId" appears 5 times in this file; the first occurrence is on line 85](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5093298)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.business'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5093298)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 7]
```diff
@@ -52,6 +51,74 @@ public static void prepare() throws Exception {
...
+
+        this.addPermission(role, host, false);
+        final List<Rule> rules = rulesAPI.getAllRulesByParent(host, user, false);
+
+        assertEquals(2, rules.size());
+
+        final List<String> rulesId = rules.stream().map(rule -> rule.getId()).collect(Collectors.toList());
+        assertTrue(rulesId.contains(rule1.getId()));
+        assertTrue(rulesId.contains(rule2.getId()));
+        assertFalse(rulesId.contains(rule3.getId()));
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: let's add a finally to remove created stuff

**Pull request merged**